torch/csrc/jit/autodiff.cpp

```diff
@@ -331,17 +331,25 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
       }
       return {sizes.at(dim) > 1 ? grads.at(0) : grads.at(0).unsqueeze(dim), nullptr};
 
-    } else if (node->matches("aten::cat(Tensor[] tensors, int dim) -> Tensor", /*const=*/attr::dim)) {
+    } else if (
```

Comment:
It looks like aten::cat was never enabled so this code doesn't do anything. You can enable it by adding an entry in differentiableops: https://github.com/pytorch/pytorch/blob/a72603f8f8ffdd77a826129b18e839df3fe3a109/torch/csrc/jit/autodiff.cpp#L27

Reply:
In addition, `aten::cat(Tensor[] tensors, int dim, Scalar pad_value)` and `"aten::cat(Tensor[] tensors, int dim) -> Tensor"` are two different schemas so you should remember to handle both paths.